PWGJE/Tasks/kstarInJets.cxx

```diff
@@ -0,0 +1,1161 @@
+  using myCompleteJetTracks = soa::Join<aod::JTracks, aod::JTrackPIs, aod::McTrackLabels>;
+  int nJEEvents = 0;
+  int nprocessRecEvents = 0;
+  void processRec(o2::aod::JCollision const& collision, myCompleteJetTracks const& tracks, soa::Filtered<aod::ChargedMCDetectorLevelJets> const& mcdjets, aod::McParticles const&, myCompleteTracks const& /*originalTracks*/)
+  {
+    if (cDebugLevel > 0) {
+      nprocessRecEvents++;
+      if ((nprocessRecEvents + 1) % 10000 == 0)
+        std::cout << "processRec: " << nprocessRecEvents << std::endl;
+    }
+
+    JEhistos.fill(HIST("nEvents_MCRec"), 0.5);
+    if (fabs(collision.posZ()) > cfgVtxCut)
+      return;
+    if (!jetderiveddatautilities::selectCollision(collision, jetderiveddatautilities::JCollisionSel::sel8))
+      return;
+
+    bool INELgt0 = false;
+    for (const auto& track : tracks) {
+      if (fabs(track.eta()) < cfgtrkMaxEta) {
+        INELgt0 = true;
+        break;
+      }
+    }
+    if (!INELgt0)
+      return;
+
+    JEhistos.fill(HIST("nEvents_MCRec"), 1.5);
+
+    std::vector<double> mcd_pt{};
+    std::vector<double> mcd_phi{};
+    std::vector<double> mcd_eta{};
+
+    bool hasJets = kFALSE;
+    for (auto& mcdjet : mcdjets) {
+      hasJets = kTRUE;
+      mcd_pt.push_back(mcdjet.pt());
+      mcd_eta.push_back(mcdjet.eta());
+      mcd_phi.push_back(mcdjet.phi());
+      registry.fill(HIST("h_jet_pt"), mcdjet.pt());
+      registry.fill(HIST("h_jet_eta"), mcdjet.eta());
+      registry.fill(HIST("h_jet_phi"), mcdjet.phi());
+    }
+    if (hasJets)
+      JEhistos.fill(HIST("nEvents_MCRec"), 2.5);
+
+    // Track Eff
+    for (const auto& track : tracks) {
+      auto originalTrack = track.track_as<myCompleteTracks>();
+      if (!trackSelection(originalTrack))
+        continue;
+      if (cfgSimPID) {
+        if (!trackPIDKaon(originalTrack))
+          continue;
+        if (!trackPIDPion(originalTrack))
+          continue;
+      }
+      if (track.has_mcParticle()) {
+        auto mcParticle = track.mcParticle();
+
+        if (mcParticle.isPhysicalPrimary() && fabs(mcParticle.eta()) <= cfgtrkMaxEta) {
+          if (abs(mcParticle.pdgCode()) == 211)
+            JEhistos.fill(HIST("ptJEHistogramPion"), mcParticle.pt());
+          if (abs(mcParticle.pdgCode()) == 321)
+            JEhistos.fill(HIST("ptJEHistogramKaon"), mcParticle.pt());
+          if (abs(mcParticle.pdgCode()) == 2212)
+            JEhistos.fill(HIST("ptJEHistogramProton"), mcParticle.pt());
+        }
+      }
+      for (const auto& track2 : tracks) {
+        auto originalTrack2 = track2.track_as<myCompleteTracks>();
+        if (!trackSelection(originalTrack2))
+          continue;
+        if (cfgSimPID) {
+          if (!trackPIDKaon(originalTrack))
+            continue;
+          if (!trackPIDPion(originalTrack))
+            continue;
+        }
+
+        if (originalTrack.index() >= originalTrack2.index())
```

Comment:
all of these should be globalIndex() ?

Reply:
I think you should reformaulate the way you do the for loops. So that the second for loop over tracks starts from the track index after the current one in the first loop